generic/tests/ethtool.py

```diff
@@ -50,7 +51,7 @@ def ethtool_get(session):
 
         o = session.cmd("ethtool -k %s" % ethname)
         status = {}
-        for f in feature_pattern.keys():
+        for f in list(feature_pattern.keys()):
```

Comment:
The same conclusion as the range() one, it can work without the change:

```python
Python 2.7.15 (default, Oct 15 2018, 15:26:09) 
[GCC 8.2.1 20180801 (Red Hat 8.2.1-2)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> d = {1: 1, 2: 2, 3: 3}
>>> for k in d.keys():
...     print(k)
... 
1
2
3
```
```python
Python 3.7.3 (default, Mar 27 2019, 13:41:07) 
[GCC 8.3.1 20190223 (Red Hat 8.3.1-2)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> d = {1: 1, 2: 2, 3: 3}
>>> for k in d.keys():
...     print(k)
... 
1
2
3
```

Reply:
@luckyh Ok.